### PR TITLE
Replace abandoned Laminas\Config\Config with local lightweight alternative

### PR DESCRIPTION
**Pull request by demiankatz**
The laminas-config component has been abandoned. This PR takes the first major step toward removing it from VuFind, by replacing Laminas\Config\Config with a lighter-weight VuFind\Config\Config. This new class should be phased out in favor of using arrays, but this allows us to get away from the deprecated component quickly without having to dramatically refactor large amounts of code.

TODO
- [x] Run full test suite before merging
- [x] Merge #4113 and #4115 before merging this

Other necessary steps (best handled in separate PRs):
- Eliminate use of Laminas\Config\Exception classes (see #4118)
- Eliminate use of Laminas\Config\Reader classes (TODO!)

### CODE REVIEW

**Comment by demiankatz** [comment]
All Mink tests are now passing here, but I'll leave this in draft mode until #4113 and #4115 are merged since it's best to keep those refactorings separate from the changes here for readability/simplicity.

File: module/VuFind/tests/unit-tests/src/VuFindTest/Config/ConfigTest.php
Comment (EreMaijala) [inline module/VuFind/tests/unit-tests/src/VuFindTest/Config/ConfigTest.php]: I think it would be more clear to simply add these assertions to the test method without a data provider.

**Review by EreMaijala — CHANGES_REQUESTED** [review]

File: module/VuFind/tests/unit-tests/src/VuFindTest/Config/ConfigTest.php
Comment (demiankatz) [reply module/VuFind/tests/unit-tests/src/VuFindTest/Config/ConfigTest.php]: I did this so that each would be counted as a separate test, since they don't depend on each other... but if you think making it more readable is preferable, I can refactor.

Comment (EreMaijala) [reply module/VuFind/tests/unit-tests/src/VuFindTest/Config/ConfigTest.php]: I'd value readability, but I'm okay keeping it this way.

Comment (EreMaijala) [reply module/VuFind/tests/unit-tests/src/VuFindTest/Config/ConfigTest.php]: I'll approve since everything else looks good to me and let you make the final decision about this. :)

**Review by EreMaijala — APPROVED** [review]

File: module/VuFind/tests/unit-tests/src/VuFindTest/Config/ConfigTest.php
Comment (demiankatz) [reply module/VuFind/tests/unit-tests/src/VuFindTest/Config/ConfigTest.php]: Thanks! As we discussed on Slack, I just went ahead and broke out the test class into separate methods... it's more verbose, but it's much more readable. :-)

**Pull request merged**
